sky/data/storage.py

```diff
@@ -873,6 +873,75 @@ def add_if_not_none(key: str, value: Optional[Any]):
         return config
 
 
+def format_gitignore_to_exclude_list(src_dir_path: str) -> List[str]:
+    """Returns a list of formatted excluded files read from .gitignore and
+    .git/info/exclude using git. 'git init' is ran when SRC_DIR_PATH
+    is not a git repository and removed after obtaining excluded list
+    """
+    expand_src_dir_path = os.path.expanduser(src_dir_path)
+
+    git_exclude_path = os.path.join(expand_src_dir_path, '.git/info/exclude')
+    gitignore_path = os.path.join(expand_src_dir_path, '.gitignore')
+
+    git_exclude_exists = os.path.isfile(git_exclude_path)
+    gitignore_exists = os.path.isfile(gitignore_path)
+
+    # This command outputs a list to be excluded according to .gitignore
+    # and .git/info/exclude
+    filter_cmd = f'git -C {expand_src_dir_path} status --ignored -s'
```

Comment:
Should we use `--porcelain=v1` instead of `-s` here? It promises to stay consistent across versions.

From the docs:
```
--porcelain[=<version>]
           Give the output in an easy-to-parse format for scripts. This is
           similar to the short output, but will remain stable across Git
           versions and regardless of user configuration. See below for
           details.
```

Reply:
Nice catch!